core/templates/pages/exploration-editor-page/exploration-save-and-publish-buttons/exploration-save-and-publish-buttons.component.ts

```diff
@@ -131,13 +141,28 @@ angular.module('oppia').component('explorationSaveAndPublishButtons', {
         $scope.saveIsInProcess = false;
         $scope.publishIsInProcess = false;
         $scope.loadingDotsAreShown = false;
+        $scope.explorationCanBePublished = false;
 
         UserExplorationPermissionsService.getPermissionsAsync()
           .then(function(permissions) {
-            $scope.showPublishButton = function() {
-              return permissions.canPublish && $scope.isPrivate();
-            };
+            $scope.explorationCanBePublished = permissions.canPublish;
           });
+        ctrl.directiveSubscriptions.add(
+          UserExplorationPermissionsService.onUserExplorationPermissionsFetched
+            .subscribe(
+              () => {
+                UserExplorationPermissionsService.getPermissionsAsync()
+                  .then(function(permissions) {
+                    $scope.explorationCanBePublished = permissions.canPublish;
+                    $scope.$applyAsync();
```

Comment:
Add TODO comment above to remove the applyAsync call

Reply:
@DubeySandeep The TODO comment is for `$rootScope` not for `$scope`
```
// TODO(#8521): Remove the use of $rootScope.$apply()
// once the controller is migrated to angular.
```